### ISSUE TITLE
Why were Angular and AngularJs lumped together?

### ISSUE THREAD

**Issue by BBlackwo** [StateOfJS/StateOfJS#137]
Thanks for putting together the survey, first off!

I'm wondering why Angular and AngularJs were lumped together this year? I don't think most people know that Angular 2+ is called Angular, whereas Angular 1 is referred to as AngularJs. 

My guess is that it skewed the results. For example last year 9.7% said they would not use again but this year it was 33.8%. I guess that a large chunk of those would be people who used AngularJs but have not yet used Angular.

What's your take on it? Thanks.

**Comment by SachaG** [comment]
We didn't want to keep having separate entries for both frameworks forever so it made sense to drop Angular.js at some point. After all, we don't have separate entries for React 15 and React 16, or Vue 2 and Vue 3. 

I agree the naming is confusing, but I think that'd be more a question for the Angular team than us. 

Note that currently 2016 and 2017 on the 2018 stream chart show data for Angular 2+, which is why the chart looks a bit weird. We're considering making those years show an aggregate of Angular.js and Angular 2+ instead, but there's some technical issues involved.

**Comment by BBlackwo** [comment]
@SachaG thanks for the quick response. That makes sense they shouldn't be separated forever. [AngularJs is now end of life](https://blog.angular.io/stable-angularjs-and-long-term-support-7e077635ee9c) anyways. It's just a shame if the results have been skewed and people (including the authors of the StateOfJs) think the Angular situation is worse than it actually is because of it. E.g. the statement:

>it's starting to look like only two combatants are left standing…

Would be interesting to see the aggregated Angular and AngularJs data - but that's still not perfect. The whole naming situation with Angular1 vs. Angular2+ is a mess.

Also, I've talked with others and read about similar concerns e.g. this article: [State of JavaScript 2018: Angular’s downfall and new up-and-comers](https://jaxenter.com/javascript-state-2018-152119.html)

>Actually, the numbers might not be _as_ bad as it seems. Previous versions of the State of JavaScript asked users about Angular 1 and Angular 2. In the 2018 version, they are combined into one.

**Comment by BBlackwo** [comment]
One more question: what did the survey question for Angular say?

**Comment by SachaG** [comment]
Yeah I guess I should clarify. The survey question in 2018 said "Angular". So you have two possibilities:

1. People correctly understood that "Angular" meant "Angular 2+". 
2. People misunderstood and some of them answered as if the question was about AngularJS. 

I think #2 is likely but that seems like the Angular team's responsibility, since all we did is take care to use the official framework's name like we did for every other framework mentioned in the survey. 

I suppose we could've formulated the question as "Angular (by which we mean Angular 2+, and *not* AngularJS" but again that becomes a bit ridiculous when no other framework requires this kind of clarification.

**Comment by SachaG** [comment]
What's ironic is that we thought highlighting Angular 2 for 2016 and 2017 was the most charitable thing to do, as the 2016-2017 number for AngularJS predictably look a lot worse. What we didn't anticipate is that having good numbers for 2016 and 2017 would make 2018 look bad in comparison…

Also I myself made things even less clear by saying that we stopped making the distinction, when it would've been more accurate to say that we simply dropped AngularJS from the survey.

**Comment by BBlackwo** [comment]
I think specifying that by Angular you meant Angular 2+ could've proved more accurate results, but I agree it is a bit ridiculous.

Agreed that saying you dropped AngularJS from the survey in 2018 would be more accurate. You think that change should be made? Or is it too late?

It's interesting in 2016 the survey said "Angular" and "Angular 2". Then in 2017 it said "Angular 1" and "Angular 2". Just goes to show how confusing the naming is.

**Comment by oburakevych** [comment]
@sachaG researchers who create survey have a responsibly to ensure the questions are clear and the data has been collected is not compromised or contaminated. It's ridiculous to blame it on Google... Just IMHO

**Comment by SachaG** [comment]
@BBlackwo I added this to the Angular results page, do you think it's clear enough?

> Note that unlike 2018, past editions of the survey also featured a question about AngularJS. But all data presented here concerns only Angular. 

@oburakevych I'll take the blame if it makes you happy. That still doesn't answer the issue of how we should phrase the question about Angular.
